main.py

```diff
@@ -105,23 +104,21 @@ def update_colors(self, links):
                             settings.COLOR_FIELD
                         )
 
-                    flow = flow_cls.from_dict(flow_dict, switch)
-                    switch_dict['flows'][neighbor] = flow
+                    switch_dict['flows'][neighbor] = flow_dict
```

Comment:
In this part you suppressed the flow´s object creation, where you pass the switch object as parameter.
Since you pass the flow dict to an external service (flowmanager), it can affect the call.

Reply:
No, it doesn't, because the switch's dpid is part of the call to flow manager, and flow manager receives a dict and creates the `Flow` from the dict and passes the switch.